game/scripts/vscripts/items/sonic.lua

```diff
@@ -0,0 +1,64 @@
+item_sonic = class(ItemBaseClass)
+
+LinkLuaModifier("modifier_item_sonic", "items/sonic.lua", LUA_MODIFIER_MOTION_NONE)
+LinkLuaModifier("modifier_sonic_fly", "modifiers/modifier_sonic_fly.lua", LUA_MODIFIER_MOTION_NONE)
+
+function item_sonic:GetIntrinsicModifierName()
+  return "modifier_sonic_fly"
+end
+
+function item_sonic:OnToggle()
```

Comment:
Instead of making this a toggle item, just use the existing transformation item mechanics.

You can remove this whole function and replace it with
```lua
function item_sonic:GetTransformationModifierName()
  return "modifier_sonic_fly"
end
```